src/main/java/moze_intel/projecte/network/commands/EMCCMD.java

```diff
@@ -0,0 +1,157 @@
+                .then(Commands.argument("player", EntityArgument.player())
+                    .then(Commands.argument("value", StringArgumentType.string())
+                        .executes((ctx) -> handle(ctx, ActionType.ADD))
+                    )
+                )
+            )
+            .then(Commands.literal("remove")
+                .requires(PEPermissions.COMMAND_EMC_REMOVE)
+                .then(Commands.argument("player", EntityArgument.player())
+                    .then(Commands.argument("value", StringArgumentType.string())
+                        .executes((ctx) -> handle(ctx, ActionType.REMOVE))
+                    )
+                )
+            )
+            .then(Commands.literal("set")
+                .requires(PEPermissions.COMMAND_EMC_SET)
+                .then(Commands.argument("player", EntityArgument.player())
+                    .then(Commands.argument("value", StringArgumentType.string())
+                        .executes((ctx) -> handle(ctx, ActionType.SET))
+                    )
+                )
+            )
+            .then(Commands.literal("test")
+                .requires(PEPermissions.COMMAND_EMC_TEST)
+                .then(Commands.argument("player", EntityArgument.player())
+                    .then(Commands.argument("value", StringArgumentType.string())
+                        .executes((ctx) -> handle(ctx, ActionType.TEST))
+                    )
+                )
+            )
+            .then(Commands.literal("get")
+                .requires(PEPermissions.COMMAND_EMC_GET)
+                .then(Commands.argument("player", EntityArgument.player())
+                    .executes((ctx) -> handle(ctx, ActionType.GET))
+                )
+            );
+    }
+
+    private static MutableComponent formatEMC(BigInteger emc) {
+        return MutableComponent.create(TransmutationEMCFormatter.formatEMC(emc).getContents()).withStyle(ChatFormatting.GRAY);
+    }
+
+
+
+    private static int handle(CommandContext<CommandSourceStack> ctx, ActionType action) throws CommandSyntaxException {
+        ServerPlayer player = EntityArgument.getPlayer(ctx, "player");
+        IKnowledgeProvider provider;
+        try {
+            provider = player.getCapability(PECapabilities.KNOWLEDGE_CAPABILITY).orElseThrow(() -> new IllegalStateException("Failed to get knowledge provider."));
+        } catch(IllegalStateException ignore) {
+            ctx.getSource().sendFailure(PELang.COMMAND_PROVIDER_FAIL.translateColored(ChatFormatting.RED));
+            return 0;
+        }
+        if (action == ActionType.GET) {
+            ctx.getSource().sendSuccess(PELang.COMMAND_EMC_GET_SUCCESS.translate(player.getDisplayName(), formatEMC(provider.getEmc())), true);
+            return 1;
+        }
+        String val = StringArgumentType.getString(ctx, "value");
+        @Nullable BigInteger value = null;
+
+        try {
+            value = new BigInteger(val);
+            switch (action) {
```

Comment:
This entire switch statement can be surrounded by the `if (value.compareTo(BigInteger.ZERO) < 0) {`